File: frontend/lib/models/player.dart
class PlayerModel {
  final String id;
  final String nickname;
  final int state; // 0 - Main menu; 1 - Invite pending; 2 - WaitRoom; 3 - In game

  PlayerModel({required this.id, required this.nickname, required this.state});

  factory PlayerModel.fromMap(Map<String, dynamic> map) {
    return PlayerModel(
      id: map['ID'] as String,
      nickname: map['Nickname'] as String,
      state: (map['State'] as int),
    );
  }
}

File: backend/models/player.go
package models

import (
	"github.com/google/uuid"
)

type PlayerState int

const (
	Main     PlayerState = iota // available for invite
	Pending                     // has sent or received an invite
	WaitRoom                    // is waiting in wait room
	Playing                     // in a game
)

type Player struct {
	ID       string
	Nickname string
	State    PlayerState
}

func NewPlayer(nickname string) *Player {
	return &Player{
		ID:       uuid.New().String(),
		Nickname: nickname,
		State:    Main,
	}
}
